.github/workflows/release.yml

```diff
@@ -128,6 +137,26 @@ jobs:
           cd target/wix
           mv amazon_qldb_shell*.msi ../../$release_name.msi
         
+          cd ../../
+          $complete_file_name = $release_name + ".msi"
+          
+          # Push unsigned MSI to S3
+          $put_object_response=( aws s3api put-object --bucket ${{ secrets.AWS_UNSIGNED_BUCKET }} --key ${{ secrets.AWS_KEY }} --body $complete_file_name --acl bucket-owner-full-control ) | ConvertFrom-Json 
+          $version_id = $put_object_response.VersionId
+
+          # Wait 5 seconds for job ID tag to be populated in S3 object
+          Start-Sleep -s 5
```

Comment:
Is there a better way to do this, like maybe how it's done on line 155?

Reply:
I couldn't find any S3 [APIs](https://docs.aws.amazon.com/cli/latest/reference/s3api/wait/index.html) that would poll an object's tagging.

The object is immediately put into the S3 bucket but it seems theres a delay for the signing job to be triggered and then it would tag the object with the job ID

Reply:
I don't mind the sleep, but I do mind if this fails and has to be manually restarted by an operator. It does not seem that your code handles an empty or invalid job id, nor does it retry.

Reply:
Added a for loop to retry up to 3 times. It will also wait 5 seconds between retry.

If it still can't retrieve the job ID at the end, then the job will exit with an error code and print to console the reason.